ArcGISRuntimeSDKQt_CppSamples/Maps/TakeScreenshot/TakeScreenshot.cpp

```diff
@@ -59,12 +59,15 @@ void TakeScreenshot::componentComplete()
   // Get the image provider from the QML Engine
   QQmlEngine* engine = QQmlEngine::contextForObject(this)->engine();
   m_imageProvider = dynamic_cast<MapImageProvider*>(engine->imageProvider(MapImageProvider::imageProviderId()));
+}
 
-  // Connect to the exportImageCompleted signal
-  connect(m_mapView, &MapQuickView::exportImageCompleted, this, [this](const QUuid& id, const QImage& img)
+// Q_INVOKABLE function to kick off the export image asynchronous task
+void TakeScreenshot::captureScreenshot()
+{
+  m_mapView->exportImageAsync().then(this, [this](QImage img)
   {
     // convert the QUuid into a QString
-    const QString imageId = id.toString().remove("{").remove("}");
+    const QString imageId = QUuid().createUuid().toString().remove("{").remove("}");
```

Comment:
Use [QUuid::WithoutBraces](https://doc.qt.io/qt-6/quuid.html#StringFormat-enum). It should do exactly what this code is doing manually.